src/webots/nodes/utils/WbUrl.cpp

```diff
@@ -295,10 +211,103 @@ bool WbUrl::isLocalUrl(const QString &url) {
 
 const QString WbUrl::computeLocalAssetUrl(const WbNode *node, QString url) {
   if (!WbApplicationInfo::repo().isEmpty() && !WbApplicationInfo::branch().isEmpty())
-    // when streaming locally, build the url from branch.txt
+    // when streaming locally, build the URL from branch.txt
     return url.replace(
       "webots://", "https://raw.githubusercontent.com/" + WbApplicationInfo::repo() + "/" + WbApplicationInfo::branch() + "/");
   else
     // when streaming a release (or nightly), "webots://" urls must be inferred
     return WbUrl::computePath(node, "url", url, false);
 }
+
+const QString WbUrl::computePrefix(const QString &rawUrl) {
+  const QString url = rawUrl.startsWith(WbNetwork::instance()->cacheDirectory()) ?
+                        WbNetwork::instance()->getUrlFromEphemeralCache(rawUrl) :
+                        rawUrl;
+
+  if (isWeb(url)) {
+    QRegularExpression re(remoteWebotsAssetRegex(true));
+    QRegularExpressionMatch match = re.match(url);
+    if (match.hasMatch())
+      return match.captured(0);
+  }
+
+  return QString();
+}
+
+const QString WbUrl::remoteWebotsAssetRegex(bool capturing) {
+  static QString regex = "https://raw.githubusercontent.com/cyberbotics/webots/[a-zA-Z0-9\\_\\-\\+]+/";
+  return capturing ? "(" + regex + ")" : regex;
+}
+
+const QString &WbUrl::remoteWebotsAssetPrefix() {
+  static QString url;
+  static bool firstCall = true;
+  if (firstCall) {
+    // if it's an official release, use the tag (for example R2022b), if it's a nightly or local distribution use the commit
+    url = "https://raw.githubusercontent.com/cyberbotics/webots/" +
+          (WbApplicationInfo::commit().isEmpty() ? WbApplicationInfo::version().toString() : WbApplicationInfo::commit()) + "/";
+    firstCall = false;
+  }
```

Comment:
This could also be simplified
```suggestion
  static QString url;
  if (url.isEmpty())
    // if it's an official release, use the tag (for example R2022b), if it's a nightly or local distribution use the commit
    url = "https://raw.githubusercontent.com/cyberbotics/webots/" +
          (WbApplicationInfo::commit().isEmpty() ? WbApplicationInfo::version().toString() : WbApplicationInfo::commit()) + "/";
```